src/gt4py/next/type_system_2/types.py

```diff
@@ -0,0 +1,257 @@
+from . import traits
+import dataclasses
+from typing import Optional, Any, Sequence
+
+from .traits import FunctionArgument
+
+
+class Type:
+    ...
+
+
+@dataclasses.dataclass
+class IntegerType(
```

Comment:
just leaving it here as a possible extension: instead of having separate types used in the type system and at runtime, we could merge this stuff with the (d)types defined in `_core.definitions`.

Reply:
I don't know the purpose of the types in `_core.definitions`, so I don't have an opinion on this.